### ISSUE TITLE
[Help Wanted] Moves seemingly never tried

### ISSUE THREAD

**Issue by ms609** [revbayes/revbayes#521]
**Describe the desired behavior.**
I'm running an MCMC analysis and noticing that almost no moves are being accepted.  

It's not impossible that this reflects an underlying issue with my model.  What strikes me as odd, however, is that certain move combinations are never being tried, despite high weights.

**What is going wrong?**

Here's an extract from my script:
```rb
moves.append( mvScale(tree_length, tuneTarget = 0.26) )
moves.append( mvScale( u, lambda = 1, weight = 20.0, tuneTarget = 0.26) )
# [...]
mymc3 = mcmcmc(mymodel, monitors, moves, nruns = 2, nchains = 6, tuneHeat = FALSE, 
  swapMethod = "both", swapMode = "multiple", combine = "mixed")
mymc3.run(generations = 40000, tuningInterval = 2000)
mymc3.operatorSummary()
```

The run is going hundreds of generations without accepting any proposals.  Here's the operator summary for chain 1; the picture's basically the same for the other chains too.

```
                  Name                  | Param              |  Weight  |  Tried   | Accepted | Acc. Ratio| Parameters
===============================================================================================================================
NNI                                      topology                14.5000      28791         16      0.0006
SubtreePruneRegraft                      topology                 2.9000       5800          4      0.0007
Scaling                                  tree_length              1.0000          0          0      0.0000 lambda = 0.0010
BetaSimplex                              rel_br_lengths          29.0000          0          0      0.0000 alpha = 100.0000
DirichletSimplex                         rel_br_lengths           2.9000          0          0      0.0000 alpha = 100.0000
Scaling                                  u                       20.0000          0          0      0.0000 lambda = 0.0010
```

I'd like to know what to make of the zeroes in the "Tried" column.  It looks like only tree rearrangements are being attempted, and other moves are ignored, despite being in the move schedule.  The "Weight" corresponds to the originally specified weights, which I presume means that the weights have not been subjected to tuning, though the alpha/lambda values have been.  I expected the values of "Tried" to have the same(ish) ratio as the values of "Weight".

**Computer info**
Win 10, RevBayes 1.2.5

**Comment by bjoelle** [comment]
could you upload the whole script so we can try it ourselves ? thanks !

**Comment by ms609** [comment]
I'll send a copy by e-mail.

**Comment by bjoelle** [comment]
ok I think it's an issue with the printing, not with the running - it is trying (and even accepting) those moves, but the counters used for printing are `num_tried_current_period` and `num_accepted_current_period` which are getting reset to 0 somewhere

**Comment by bjoelle** [comment]
there is a flag called `current_period` in the printing which is set to `false` but it doesn't actually control anything in MCMCMC - it only works for the regular MCMC

**Comment by bjoelle** [comment]
ok the counters are getting reset when tuning - so I think this is why NNI and SPR are fine, because they don't have a tuning parameter so can't be tuned

**Comment by bjoelle** [comment]
so the main issue seems to be that the `current_period` flag does not work properly for MCMCMC and so the default printing behaviour is different between MCMC and MCMCMC
if the flag worked it would show `num_tried_total` instead and that is > 0

@ms609 I don't have time to work on this more right now but I believe the underlying analysis is doing the right thing, it's the summary which is wrong - on a longer analysis you should be able to check that e.g. `u` is actually moving

**Comment by ms609** [comment]
Thanks for looking into this, @bjoelle .  This goes some way to explaining the behaviour.  

Even if it is not quite zero, I still think that the acceptance rate is real – `u` moves perhaps two or three times over 40k generations, moving less frequently as the run proceeds; and the extreme alpha and lambda values suggest that tuning is favouring tiny moves, presumably in a desperate attempt to increase the acceptance ratio.


Even the NNI / SPR acceptance rates feel incredibly small to me.

**Comment by bjoelle** [comment]
is it possible that it's getting stuck in a weird configuration ? it seemed to move OK in my trials, but I only did like 1000 iterations

**Comment by ms609** [comment]
That's not impossible – indeed, given that the model's a bit quirky and I've not checked the maths in detail, it's quite possible that it's displaying unhelpful behaviour.  It does seem to get stuck very quickly (typically within a couple of thousand moves), and to get stuck in a different place each time I run the script.  The odd thing would be if it wasn't getting stuck for you – which I suppose could be explained by your short run duration?

**Comment by bjoelle** [comment]
in my runs `u` very quickly goes to 0 (e-16 after 500 iterations) so there seems to be something going on there but without knowing anything about the model it's hard to know if that is intended behaviour

**Comment by davidcerny** [comment]
Fixed by PR #522.
